### DISCUSSION TITLE
How to change already existing database table image field to directus image field

### DISCUSSION THREAD

**Discussion by harshalone** [directus/directus#14408] [Archive]
How to change already existing database table image field to directus image field?

The current image field is just a type : string to save the url of the image. I want to change it to the directus image field so that I can upload image to GCS storage and then store the URL into that field.

**Comment by rijkvanzanten** [comment]
That won't work like that. Directus doesn't store absolutely URLs to files when using the files interface, so those two approaches aren't directly compatible. Directus relies on storing the files UUID as saved in directus_files for the interfaces 👍🏻

**Reply by benhaynes** [reply]
That being said, I do think we should have an interface that displays a URL value as media. That's a common way to store references to external assets.

**Reply by rijkvanzanten** [reply]
That's gonna be pretty tricky to configure though, as security rules prevent any old media to be embedded on the page (as that's a vector for XSS attacks). One would have to explicitly white-list every domain that can be used to embed the media in the environment variables first
